django/contrib/auth/models.py

```diff
@@ -199,16 +202,24 @@ def _user_get_permissions(user, obj, from_name):
     return permissions
 
 
-def _user_has_perm(user, perm, obj):
+def _user_has_perm(user, perm, obj, **kwargs):
     """
     A backend can raise `PermissionDenied` to short-circuit permission checking.
     """
     for backend in auth.get_backends():
         if not hasattr(backend, 'has_perm'):
             continue
         try:
-            if backend.has_perm(user, perm, obj):
-                return True
+            try:
+                if backend.has_perm(user, perm, obj, **kwargs):
+                    return True
+            except TypeError:
+                warnings.warn(
+                    "Backend has_perm should accept also **kwargs",
+                    RemovedInDjango40Warning, stacklevel=2
```

Comment:
We're warning in 4.0 but removing in 5.0, so it should be `RemovedInDjango50Warning`. Search the Django docs for Deprecating a feature, which has tips for other docs that need updating like `docs/internals/deprecation.txt`